[CI] SecurityNetty4HttpServerTransportTests testMalformedRequestDispatchedNoAuthn failing

Introduced by #95886. We've got a turkish I here causing string equality mismatch

**Build scan:**
https://gradle-enterprise.elastic.co/s/iik5gi62i5sdq/tests/:x-pack:plugin:security:test/org.elasticsearch.xpack.security.transport.netty4.SecurityNetty4HttpServerTransportTests/testMalformedRequestDispatchedNoAuthn

**Reproduction line:**
```
./gradlew ':x-pack:plugin:security:test' --tests "org.elasticsearch.xpack.security.transport.netty4.SecurityNetty4HttpServerTransportTests.testMalformedRequestDispatchedNoAuthn" -Dtests.seed=A4B64B6F13AB9626 -Dtests.locale=tr -Dtests.timezone=US/Indiana-Starke -Druntime.java=20
```

**Applicable branches:**
main

**Reproduces locally?:**
Yes

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.xpack.security.transport.netty4.SecurityNetty4HttpServerTransportTests&tests.test=testMalformedRequestDispatchedNoAuthn

**Failure excerpt:**
```
java.lang.AssertionError: 
Expected: a string containing "NOT A VALID HTTP LINE"
     but: was "java.lang.IllegalArgumentException: invalid version format: NOT A VALİD HTTP LİNE"

  at __randomizedtesting.SeedInfo.seed([A4B64B6F13AB9626:D95AF3108B934C8B]:0)
  at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
  at org.junit.Assert.assertThat(Assert.java:956)
  at org.junit.Assert.assertThat(Assert.java:923)
  at org.elasticsearch.xpack.security.transport.netty4.SecurityNetty4HttpServerTransportTests.testMalformedRequestDispatchedNoAuthn(SecurityNetty4HttpServerTransportTests.java:584)
  at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
  at java.lang.reflect.Method.invoke(Method.java:578)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:1623)

```

--- a/test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java
+++ b/test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java
@@ -2002,4 +2002,9 @@ public abstract class ESTestCase extends LuceneTestCase {
             throw new AssertionError("unexpected", e);
         }
     }
+
+    protected static boolean isTurkishLocale() {
+        return Locale.getDefault().getLanguage().equals(new Locale("tr").getLanguage())
+            || Locale.getDefault().getLanguage().equals(new Locale("az").getLanguage());
+    }
 }

--- a/x-pack/plugin/identity-provider/src/test/java/org/elasticsearch/xpack/idp/saml/test/IdpSamlTestCase.java
+++ b/x-pack/plugin/identity-provider/src/test/java/org/elasticsearch/xpack/idp/saml/test/IdpSamlTestCase.java
@@ -76,11 +76,6 @@ public abstract class IdpSamlTestCase extends ESTestCase {
         }
     }
 
-    private static boolean isTurkishLocale() {
-        return Locale.getDefault().getLanguage().equals(new Locale("tr").getLanguage())
-            || Locale.getDefault().getLanguage().equals(new Locale("az").getLanguage());
-    }
-
     @AfterClass
     public static void restoreLocale() {
         if (restoreLocale != null) {

--- a/x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/saml/SamlTestCase.java
+++ b/x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/saml/SamlTestCase.java
@@ -53,11 +53,6 @@ public abstract class SamlTestCase extends ESTestCase {
         SamlUtils.initialize(logger);
     }
 
-    private static boolean isTurkishLocale() {
-        return Locale.getDefault().getLanguage().equals(new Locale("tr").getLanguage())
-            || Locale.getDefault().getLanguage().equals(new Locale("az").getLanguage());
-    }
-
     @AfterClass
     public static void restoreLocale() {
         if (restoreLocale != null) {

--- a/x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/transport/netty4/SecurityNetty4HttpServerTransportTests.java
+++ b/x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/transport/netty4/SecurityNetty4HttpServerTransportTests.java
@@ -526,7 +526,7 @@ public class SecurityNetty4HttpServerTransportTests extends AbstractHttpServerTr
 
     public void testMalformedRequestDispatchedNoAuthn() throws Exception {
         assumeTrue(
-            "This test doesn't work correctly under turkish-like locale, " + "because it uses String#toUpper() for asserted error messages",
+            "This test doesn't work correctly under turkish-like locale, because it uses String#toUpper() for asserted error messages",
             !isTurkishLocale()
         );
         final AtomicReference<Throwable> dispatchThrowableReference = new AtomicReference<>();

--- a/x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/transport/netty4/SecurityNetty4HttpServerTransportTests.java
+++ b/x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/transport/netty4/SecurityNetty4HttpServerTransportTests.java
@@ -527,7 +527,7 @@ public class SecurityNetty4HttpServerTransportTests extends AbstractHttpServerTr
     public void testMalformedRequestDispatchedNoAuthn() throws Exception {
         assumeTrue(
             "This test doesn't work correctly under turkish-like locale, because it uses String#toUpper() for asserted error messages",
-            !isTurkishLocale()
+            isTurkishLocale() == false
         );
         final AtomicReference<Throwable> dispatchThrowableReference = new AtomicReference<>();
         final AtomicInteger authnInvocationCount = new AtomicInteger();

--- a/x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/transport/netty4/SecurityNetty4HttpServerTransportTests.java
+++ b/x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/transport/netty4/SecurityNetty4HttpServerTransportTests.java
@@ -525,6 +525,10 @@ public class SecurityNetty4HttpServerTransportTests extends AbstractHttpServerTr
     }
 
     public void testMalformedRequestDispatchedNoAuthn() throws Exception {
+        assumeTrue(
+            "This test doesn't work correctly under turkish-like locale, " + "because it uses String#toUpper() for asserted error messages",
+            !isTurkishLocale()
+        );
         final AtomicReference<Throwable> dispatchThrowableReference = new AtomicReference<>();
         final AtomicInteger authnInvocationCount = new AtomicInteger();
         final AtomicInteger badDispatchInvocationCount = new AtomicInteger();
@@ -574,8 +578,8 @@ public class SecurityNetty4HttpServerTransportTests extends AbstractHttpServerTr
                 EmbeddedChannel ch = new EmbeddedChannel(handler);
                 ByteBuf buf = ch.alloc().buffer();
                 ByteBufUtil.copy(AsciiString.of("This is not a valid HTTP line"), buf);
-                ByteBufUtil.writeShortBE(buf, HttpConstants.LF);
-                ByteBufUtil.writeShortBE(buf, HttpConstants.LF);
+                buf.writeByte(HttpConstants.LF);
+                buf.writeByte(HttpConstants.LF);
                 var writeFuture = testThreadPool.generic().submit(() -> {
                     ch.writeInbound(buf);
                     ch.flushInbound();
@@ -590,8 +594,8 @@ public class SecurityNetty4HttpServerTransportTests extends AbstractHttpServerTr
                 EmbeddedChannel ch = new EmbeddedChannel(handler);
                 ByteBuf buf = ch.alloc().buffer();
                 ByteBufUtil.copy(AsciiString.of("GET /this/is/a/valid/but/too/long/initial/line HTTP/1.1"), buf);
-                ByteBufUtil.writeShortBE(buf, HttpConstants.LF);
-                ByteBufUtil.writeShortBE(buf, HttpConstants.LF);
+                buf.writeByte(HttpConstants.LF);
+                buf.writeByte(HttpConstants.LF);
                 var writeFuture = testThreadPool.generic().submit(() -> {
                     ch.writeInbound(buf);
                     ch.flushInbound();
@@ -606,10 +610,10 @@ public class SecurityNetty4HttpServerTransportTests extends AbstractHttpServerTr
                 EmbeddedChannel ch = new EmbeddedChannel(handler);
                 ByteBuf buf = ch.alloc().buffer();
                 ByteBufUtil.copy(AsciiString.of("GET /url HTTP/1.1"), buf);
-                ByteBufUtil.writeShortBE(buf, HttpConstants.LF);
+                buf.writeByte(HttpConstants.LF);
                 ByteBufUtil.copy(AsciiString.of("Host"), buf);
-                ByteBufUtil.writeShortBE(buf, HttpConstants.LF);
-                ByteBufUtil.writeShortBE(buf, HttpConstants.LF);
+                buf.writeByte(HttpConstants.LF);
+                buf.writeByte(HttpConstants.LF);
                 var writeFuture = testThreadPool.generic().submit(() -> {
                     ch.writeInbound(buf);
                     ch.flushInbound();
@@ -624,10 +628,10 @@ public class SecurityNetty4HttpServerTransportTests extends AbstractHttpServerTr
                 EmbeddedChannel ch = new EmbeddedChannel(handler);
                 ByteBuf buf = ch.alloc().buffer();
                 ByteBufUtil.copy(AsciiString.of("GET /url HTTP/1.1"), buf);
-                ByteBufUtil.writeShortBE(buf, HttpConstants.LF);
+                buf.writeByte(HttpConstants.LF);
                 ByteBufUtil.copy(AsciiString.of("Host: this.looks.like.a.good.url.but.is.longer.than.permitted"), buf);
-                ByteBufUtil.writeShortBE(buf, HttpConstants.LF);
-                ByteBufUtil.writeShortBE(buf, HttpConstants.LF);
+                buf.writeByte(HttpConstants.LF);
+                buf.writeByte(HttpConstants.LF);
                 var writeFuture = testThreadPool.generic().submit(() -> {
                     ch.writeInbound(buf);
                     ch.flushInbound();
@@ -642,10 +646,11 @@ public class SecurityNetty4HttpServerTransportTests extends AbstractHttpServerTr
                 EmbeddedChannel ch = new EmbeddedChannel(handler);
                 ByteBuf buf = ch.alloc().buffer();
                 ByteBufUtil.copy(AsciiString.of("GET /url HTTP/1.1"), buf);
-                ByteBufUtil.writeShortBE(buf, HttpConstants.LF);
-                ByteBufUtil.copy(AsciiString.of("Host: invalid host value"), buf);
-                ByteBufUtil.writeShortBE(buf, HttpConstants.LF);
-                ByteBufUtil.writeShortBE(buf, HttpConstants.LF);
+                buf.writeByte(HttpConstants.LF);
+                ByteBufUtil.copy(AsciiString.of("Host: invalid header value"), buf);
+                buf.writeByte(0x01);
+                buf.writeByte(HttpConstants.LF);
+                buf.writeByte(HttpConstants.LF);
                 var writeFuture = testThreadPool.generic().submit(() -> {
                     ch.writeInbound(buf);
                     ch.flushInbound();
@@ -660,9 +665,9 @@ public class SecurityNetty4HttpServerTransportTests extends AbstractHttpServerTr
                 EmbeddedChannel ch = new EmbeddedChannel(handler);
                 ByteBuf buf = ch.alloc().buffer();
                 ByteBufUtil.copy(AsciiString.of("GET /url HTTP/1.1"), buf);
-                ByteBufUtil.writeShortBE(buf, HttpConstants.LF);
+                buf.writeByte(HttpConstants.LF);
                 ByteBufUtil.copy(AsciiString.of("Host: localhost"), buf);
-                ByteBufUtil.writeShortBE(buf, HttpConstants.LF);
+                buf.writeByte(HttpConstants.LF);
                 testThreadPool.generic().submit(() -> {
                     ch.writeInbound(buf);
                     ch.flushInbound();

